Template/Variable/CustomJsFunctionVariable.php

```diff
@@ -36,30 +36,42 @@ public function getParameters()
                 $field->title = 'JavaScript Function';
                 $field->description = 'The value should start with "function() { " and end with "return yourValue; }". You have to define a function and return a value. We highly recommend to test the pasted JavaScript function to avoid JavaScript errors on your website.';
                 $field->uiControl = FieldConfig::UI_CONTROL_TEXTAREA;
+                $field->customFieldComponent = self::FIELD_TEXTAREA_VARIABLE_COMPONENT;
+                $field->prepare = function ($value) {
```

Comment:
@ulcuber any reason why have we added this line ?

Reply:
Added because value was trimmed twice

Reply:
I don't think `prepare` works in anyway, you can simply added the trim back may be ?

Reply:
In which case should it not work?

Reply:
@ulcuber Unable to find any implementation of `prepare` in `https://github.com/matomo-org/matomo/blob/4.x-dev/core/Settings/FieldConfig.php`

Reply:
https://github.com/matomo-org/matomo/blob/48c458c830fd75ab8cc90489d5a31110ee6c36c9/core/Settings/FieldConfig.php#L213
https://github.com/matomo-org/matomo/blob/48c458c830fd75ab8cc90489d5a31110ee6c36c9/core/Settings/Setting.php#L224

Reply:
Hey @ulcuber Thanks for pointing it out, my IDE was not recognising the `prepare` changes dunno why.
I can see now :+1: